projects/BankIDAuth/code/deploy-contact-flow/lambda_function.py

```diff
@@ -0,0 +1,144 @@
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# Supports Amazon Connect Instances from any region
+client = boto3.client('connect', region_name=CONNECT_REGION)
+
+
+def lambda_handler(event, context):
+    logger.info('Creating contact flow.')
+    logger.info({ 'event': event })
+
+    queue_arn = fetch_sample_basic_queue_arn()
+    template = load_content(CONTACT_FLOW)
+    content = update_content(template, queue_arn)
+
+    try:
+        # Gracefully handle re-deployments
+        contact_flow = fetch_contact_flow()
+        if contact_flow:
+            # Contact Flow was already created
+            resp_data = update_contact_flow(content, contact_flow)
+        else:
+            # Contact Flow does not exist
+            resp_data = create_contact_flow(content)
+
+        logger.debug({ 'responseData': resp_data })
+        cfnresponse.send(event, context, cfnresponse.SUCCESS, resp_data)
+
+    except ClientError as e:
+        error = e.response['Error']
+        logger.error({ 'Error': error })
+        cfnresponse.send(event, context, cfnresponse.FAILED, error)
+
+
+def create_contact_flow(content):
+    resp = client.create_contact_flow(
+        InstanceId=INSTANCE_ID,
+        Name=CONTACT_FLOW_NAME,
+        Type='CONTACT_FLOW',
+        Description='Verifies the callers identity with BankID integration.',
+        Content=content
+    )
+
+    logger.info('Contact Flow created.')
+    return {
+        'Arn': resp['ContactFlowArn']
+    }
+
+def update_contact_flow(content, contact_flow):
+    client.update_contact_flow_content(
+        InstanceId=INSTANCE_ID,
+        ContactFlowId=contact_flow['Id'],
+        Content=content
+    )
+
+    logger.info('Contact Flow updated.')
+    return {
+        'Arn': contact_flow['Arn']
+    }
+
+def load_content(fn):
+    # Loads contact flow content from json file
+    logger.info('Loading content from %s', fn)
+    with open(fn, encoding='utf-8') as f:
+        content = json.load(f)
+
+    logger.debug({ 'content': content })
+    return content
+
+
+def fetch_sample_basic_queue_arn():
+    # All Amazon Connect instances have a Sample Basic Queue
+    # This function collects the Arn for the Sample Basic Queue
+    # Which will be used in the contact flow to connect the caller
+    # to an agent.
+    logger.info('Fetching Sample Basic Queue Arn')
+    response = client.list_queues(
+        InstanceId=INSTANCE_ID,
+        QueueTypes=['STANDARD']
+    )
+
+    logger.debug({ 'response': response })
+    for i in response['QueueSummaryList']:
+        if i['Name'] == 'Sample BasicQueue':
+            logger.info({ 'Arn': i['Arn'] })
+            return i['Arn']
+
+
+def fetch_contact_flow():
+    # Gracefully handle duplication errors during re-deployment
+    # This function collects the info for the Contact Flow
+    # So that the contact flow can be updated instead of created
+    logger.info('Fetching Contact Flow ID')
+    response = client.list_contact_flows(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `list_contact_flows` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)